src/api/hooks/useFetchCurrentUser.tsx

```diff
@@ -1,17 +1,16 @@
-import { useMemo } from 'react'
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { apiUrl } from '@config/apiUrl'
 import { User } from '@api/interfaces'
 
 export const useFetchCurrentUser = () => {
-  const userData = useMemo<User>(
-    () => ({
-      id: 'id',
-      username: 'jeicaM',
-      email: 'json.connor@wp.com',
-      avatarUrl: '/avatar.png'
-      // TODO: These should be generated on backend
-    }),
-    []
+  return useQuery(
+    [],
+    () => {
+      return axios
+        .get<User>(`${apiUrl}users`, { withCredentials: true })
```

Comment:
In axios, you can create a client and configure it's defaults. It would be helpful do use it, because you wouldn't have to pass in baseURL or options like withCredentials everytime you use it. You can also set other useful details. https://axios-http.com/docs/instance